<html><h1>Validation FAILED.</h1><p>Build Time: 30 minutes 55 seconds 462 ms</p><h4>Base Branch:</h4><p>Branch Name: <a href="https://github.com/liferay/com-liferay-commerce/tree/7.1.x">7.1.x</a><br/>Branch GIT ID: <a href="https://github.com/liferay/com-liferay-commerce/commit/3ce635d1bdfa2bbbfa98a449dd5b6ef796be9801">3ce635d1bdfa2bbbfa98a449dd5b6ef796be9801</a></p><h6>Task Summary:</h6><ul><li>subrepository-compile-jdk8 -  :heavy_check_mark:</li><li>subrepository-semantic-versioning-jdk8 -  :heavy_check_mark:</li><li>subrepository-pmd-jdk8 -  :heavy_check_mark:</li><li>subrepository-source-format-jdk8 -  :x:<pre><code>     [echo] java.lang.Exception: Found 1 formatting issues:
     [echo] 1: ./commerce-product-service/src/main/java/com/liferay/commerce/product/search/CPDefinitionIndexer.java expected:&lt;...&gt; availableLocales =[ LanguageUtil.getAvailableLocales(
     [echo] ]				cpDefinitionSpec...&gt; but was:&lt;...&gt; availableLocales =[
     [echo] 				LanguageUtil.getAvailableLocales(
     [echo] 	]				cpDefinitionSpec...&gt;
     [echo] 
     [echo] 	at com.liferay.source.formatter.SourceFormatter.format(SourceFormatter.java:430)
     [echo] 	at com.liferay.source.formatter.SourceFormatter.main(SourceFormatter.java:258)
     [echo] 
     [echo] FAILURE: Build failed with an exception.
     [echo] 
     [echo] * What went wrong:
     [echo] Execution failed for task ':checkSourceFormatting'.
     [echo] &gt; Process 'command '/opt/java/jdk1.8.0_121-x64/bin/java'' finished with non-zero exit value 1
     [echo] 
     [echo] * Try:
     [echo] Run with --info or --debug option to get more log output. Run with --scan to get full insights.
     [echo] 
     [echo] * Exception is:
     [echo] org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':checkSourceFormatting'.
     [echo] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:110)
     [echo] 	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:77)
     [echo] 	at org.gradle.api.internal.tasks.execution.OutputDirectoryCreatingTaskExecuter.execute(OutputDirectoryCreatingTaskExecuter.java:51)
     [echo] 	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:59)
     [echo] 	at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
     [echo] 	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:59)
     [echo] 	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:101)
     [echo] 	at org.gradle.api.internal.tasks.execution.FinalizeInputFilePropertiesTaskExecuter.execute(FinalizeInputFilePropertiesTaskExecuter.java:44)
     [echo] 	at org.gradle.api.internal.tasks.execution.CleanupStaleOutputsExecuter.execute(CleanupStaleOutputsExecuter.java:91)</code></pre></li><li>central-requirements-jdk8/subrepository-unit-jdk8 -  :heavy_check_mark:</li></ul><h5>For full console, click <a href="https://test-1-6.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/47//consoleText">here</a>.</h5><hr/><div><h6>Test Results:</h6><div><div><p>27 Tests Passed.<br/>0 Tests Failed.</p></div></div><h5>For all test results, click <a href="https://test-1-6.liferay.com/job/test-subrepository-acceptance-pullrequest-validation(7.1.x)/47//testReport">here</a>.</h5></div></html>

--- a/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPSpecificationOptionFacetsDisplayContext.java
+++ b/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPSpecificationOptionFacetsDisplayContext.java
@@ -15,14 +15,13 @@
 package com.liferay.commerce.product.content.search.web.internal.display.context;
 
 import com.liferay.commerce.product.content.search.web.internal.util.CPSpecificationOptionFacetsUtil;
-import com.liferay.commerce.product.model.CPDefinitionSpecificationOptionValue;
 import com.liferay.commerce.product.model.CPSpecificationOption;
 import com.liferay.commerce.product.service.CPDefinitionSpecificationOptionValueService;
 import com.liferay.commerce.product.service.CPSpecificationOptionLocalService;
 import com.liferay.portal.kernel.exception.PortalException;
 import com.liferay.portal.kernel.search.facet.Facet;
 import com.liferay.portal.kernel.util.ArrayUtil;
-import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
 import com.liferay.portal.search.web.portlet.shared.search.PortletSharedSearchResponse;
 
 import java.util.List;
@@ -56,21 +55,21 @@ public class CPSpecificationOptionFacetsDisplayContext {
 	public CPSpecificationOption getCPSpecificationOption(String fieldName)
 		throws PortalException {
 
-		String cpSpecificationOptionId =
+		String key =
 			CPSpecificationOptionFacetsUtil.
-				getCPSpecificationOptionIdFromIndexFieldName(fieldName);
+				getCPSpecificationOptionKeyFromIndexFieldName(fieldName);
 
 		return _cpSpecificationOptionLocalService.fetchCPSpecificationOption(
-			GetterUtil.getLong(cpSpecificationOptionId));
+			PortalUtil.getScopeGroupId(_renderRequest), key);
 	}
 
-	public long getCPSpecificationOptionKey(String fieldName)
+	public String getCPSpecificationOptionKey(String fieldName)
 		throws PortalException {
 
 		CPSpecificationOption cpSpecificationOption = getCPSpecificationOption(
 			fieldName);
 
-		return cpSpecificationOption.getCPSpecificationOptionId();
+		return cpSpecificationOption.getKey();
 	}
 
 	public String getCPSpecificationOptionTitle(String fieldName)
@@ -80,18 +79,6 @@ public class CPSpecificationOptionFacetsDisplayContext {
 			fieldName);
 
 		return cpSpecificationOption.getTitle(_locale);
-	}
-
-	public String getDisplayName(Locale locale, String key)
-		throws PortalException {
-
-		CPDefinitionSpecificationOptionValue
-			cpDefinitionSpecificationOptionValue =
-				_cpDefinitionSpecificationOptionValueService.
-					fetchCPDefinitionSpecificationOptionValue(
-						GetterUtil.getLong(key));
-
-		return cpDefinitionSpecificationOptionValue.getValue(locale);
 	}
 
 	public List<Facet> getFacets() {
@@ -107,9 +94,7 @@ public class CPSpecificationOptionFacetsDisplayContext {
 
 		Optional<String[]> parameterValuesOptional =
 			_portletSharedSearchResponse.getParameterValues(
-				String.valueOf(
-					cpSpecificationOption.getCPSpecificationOptionId()),
-				_renderRequest);
+				cpSpecificationOption.getKey(), _renderRequest);
 
 		if (parameterValuesOptional.isPresent()) {
 			String[] parameterValues = parameterValuesOptional.get();

--- a/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPSpecificationOptionFacetsDisplayContext.java
+++ b/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/display/context/CPSpecificationOptionFacetsDisplayContext.java
@@ -16,7 +16,6 @@ package com.liferay.commerce.product.content.search.web.internal.display.context
 
 import com.liferay.commerce.product.content.search.web.internal.util.CPSpecificationOptionFacetsUtil;
 import com.liferay.commerce.product.model.CPSpecificationOption;
-import com.liferay.commerce.product.service.CPDefinitionSpecificationOptionValueService;
 import com.liferay.commerce.product.service.CPSpecificationOptionLocalService;
 import com.liferay.portal.kernel.exception.PortalException;
 import com.liferay.portal.kernel.search.facet.Facet;
@@ -36,14 +35,10 @@ import javax.portlet.RenderRequest;
 public class CPSpecificationOptionFacetsDisplayContext {
 
 	public CPSpecificationOptionFacetsDisplayContext(
-		CPDefinitionSpecificationOptionValueService
-			cpDefinitionSpecificationOptionValueService,
 		CPSpecificationOptionLocalService cpSpecificationOptionLocalService,
 		RenderRequest renderRequest, List<Facet> facets,
 		PortletSharedSearchResponse portletSharedSearchResponse) {
 
-		_cpDefinitionSpecificationOptionValueService =
-			cpDefinitionSpecificationOptionValueService;
 		_cpSpecificationOptionLocalService = cpSpecificationOptionLocalService;
 		_renderRequest = renderRequest;
 		_facets = facets;
@@ -105,8 +100,6 @@ public class CPSpecificationOptionFacetsDisplayContext {
 		return false;
 	}
 
-	private final CPDefinitionSpecificationOptionValueService
-		_cpDefinitionSpecificationOptionValueService;
 	private final CPSpecificationOptionLocalService
 		_cpSpecificationOptionLocalService;
 	private final List<Facet> _facets;

--- a/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/CPSpecificationOptionFacetsPortlet.java
+++ b/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/CPSpecificationOptionFacetsPortlet.java
@@ -99,14 +99,14 @@ public class CPSpecificationOptionFacetsPortlet
 			List<Facet> facets = getFacets(renderRequest);
 
 			for (Facet facet : facets) {
-				String cpSpecificationOptionId =
+				String cpSpecificationOptionKey =
 					CPSpecificationOptionFacetsUtil.
-						getCPSpecificationOptionIdFromIndexFieldName(
+						getCPSpecificationOptionKeyFromIndexFieldName(
 							facet.getFieldName());
 
 				Optional<String[]> parameterValuesOptional =
 					portletSharedSearchSettings.getParameterValues(
-						cpSpecificationOptionId);
+						cpSpecificationOptionKey);
 
 				MultiValueFacet multiValueFacet = new MultiValueFacet(
 					searchContext);
@@ -188,6 +188,9 @@ public class CPSpecificationOptionFacetsPortlet
 	protected List<Facet> getFacets(RenderRequest renderRequest)
 		throws Exception {
 
+		ThemeDisplay themeDisplay = (ThemeDisplay)renderRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
 		List<Facet> facets = new ArrayList<>();
 
 		AssetCategory assetCategory = (AssetCategory)renderRequest.getAttribute(
@@ -202,14 +205,14 @@ public class CPSpecificationOptionFacetsPortlet
 
 		Facet facet = new SimpleFacet(searchContext);
 
-		facet.setFieldName(CPDefinitionIndexer.FIELD_SPECIFICATION_IDS);
+		facet.setFieldName(CPDefinitionIndexer.FIELD_SPECIFICATION_NAMES);
 
 		searchContext.addFacet(facet);
 
 		QueryConfig queryConfig = new QueryConfig();
 
 		queryConfig.addSelectedFieldNames(
-			CPDefinitionIndexer.FIELD_SPECIFICATION_IDS);
+			CPDefinitionIndexer.FIELD_SPECIFICATION_NAMES);
 
 		queryConfig.setHighlightEnabled(false);
 		queryConfig.setScoreEnabled(false);
@@ -229,7 +232,7 @@ public class CPSpecificationOptionFacetsPortlet
 
 			multiValueFacet.setFieldName(
 				CPSpecificationOptionFacetsUtil.getIndexFieldName(
-					termCollector.getTerm()));
+					termCollector.getTerm(), themeDisplay.getLanguageId()));
 
 			facets.add(multiValueFacet);
 		}

--- a/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/CPSpecificationOptionFacetsPortlet.java
+++ b/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/portlet/CPSpecificationOptionFacetsPortlet.java
@@ -20,7 +20,6 @@ import com.liferay.commerce.product.content.search.web.internal.display.context.
 import com.liferay.commerce.product.content.search.web.internal.util.CPSpecificationOptionFacetsUtil;
 import com.liferay.commerce.product.model.CPDefinition;
 import com.liferay.commerce.product.search.CPDefinitionIndexer;
-import com.liferay.commerce.product.service.CPDefinitionSpecificationOptionValueService;
 import com.liferay.commerce.product.service.CPSpecificationOptionLocalService;
 import com.liferay.portal.kernel.log.Log;
 import com.liferay.portal.kernel.log.LogFactoryUtil;
@@ -150,7 +149,6 @@ public class CPSpecificationOptionFacetsPortlet
 			CPSpecificationOptionFacetsDisplayContext
 				cpSpecificationOptionFacetsDisplayContext =
 					new CPSpecificationOptionFacetsDisplayContext(
-						_cpDefinitionSpecificationOptionValueService,
 						_cpSpecificationOptionLocalService, renderRequest,
 						filledFacets, portletSharedSearchResponse);
 
@@ -245,10 +243,6 @@ public class CPSpecificationOptionFacetsPortlet
 
 	private static final Log _log = LogFactoryUtil.getLog(
 		CPSpecificationOptionFacetsPortlet.class);
-
-	@Reference
-	private CPDefinitionSpecificationOptionValueService
-		_cpDefinitionSpecificationOptionValueService;
 
 	@Reference
 	private CPSpecificationOptionLocalService

--- a/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/util/CPSpecificationOptionFacetsUtil.java
+++ b/commerce-product-content-search-web/src/main/java/com/liferay/commerce/product/content/search/web/internal/util/CPSpecificationOptionFacetsUtil.java
@@ -15,6 +15,7 @@
 package com.liferay.commerce.product.content.search.web.internal.util;
 
 import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringBundler;
 import com.liferay.portal.kernel.util.StringUtil;
 import com.liferay.portal.kernel.util.Validator;
 
@@ -23,7 +24,7 @@ import com.liferay.portal.kernel.util.Validator;
  */
 public class CPSpecificationOptionFacetsUtil {
 
-	public static String getCPSpecificationOptionIdFromIndexFieldName(
+	public static String getCPSpecificationOptionKeyFromIndexFieldName(
 		String fieldName) {
 
 		if (Validator.isNull(fieldName)) {
@@ -33,11 +34,12 @@ public class CPSpecificationOptionFacetsUtil {
 		String[] fieldNameParts = StringUtil.split(
 			fieldName, StringPool.UNDERLINE);
 
-		return fieldNameParts[1];
+		return fieldNameParts[3];
 	}
 
-	public static String getIndexFieldName(String key) {
-		return "SPECIFICATION_" + key + "_VALUE_ID";
+	public static String getIndexFieldName(String key, String languageId) {
+		return StringBundler.concat(
+			languageId, "_SPECIFICATION_", key, "_VALUE_NAME");
 	}
 
 }

--- a/commerce-product-content-search-web/src/main/resources/META-INF/resources/specification_option_facets/view.jsp
+++ b/commerce-product-content-search-web/src/main/resources/META-INF/resources/specification_option_facets/view.jsp
@@ -60,7 +60,7 @@ for (Facet facet : cpSpecificationOptionFacetsDisplayContext.getFacets()) {
 							<label class="facet-checkbox-label" for="<portlet:namespace />term_<%= facet.getFieldName() + i %>">
 								<input
 									class="facet-term"
-									data-term-id="<%= termCollector.getTerm() %>"
+									data-term-id="<%= HtmlUtil.escape(termCollector.getTerm()) %>"
 									id="<portlet:namespace />term_<%= facet.getFieldName() + i %>"
 									name="<portlet:namespace />term_<%= facet.getFieldName() + i %>"
 									onChange="Liferay.Search.FacetUtil.changeSelection(event);"
@@ -69,7 +69,7 @@ for (Facet facet : cpSpecificationOptionFacetsDisplayContext.getFacets()) {
 								/>
 
 								<span class="term-name">
-									<%= HtmlUtil.escape(cpSpecificationOptionFacetsDisplayContext.getDisplayName(locale, termCollector.getTerm())) %>
+									<%= HtmlUtil.escape(termCollector.getTerm()) %>
 								</span>
 
 								<small class="term-count">

--- a/commerce-product-service/src/main/java/com/liferay/commerce/product/search/CPDefinitionIndexer.java
+++ b/commerce-product-service/src/main/java/com/liferay/commerce/product/search/CPDefinitionIndexer.java
@@ -37,6 +37,7 @@ import com.liferay.commerce.product.service.CPInstanceLocalService;
 import com.liferay.commerce.product.util.CPRulesThreadLocal;
 import com.liferay.portal.kernel.dao.orm.IndexableActionableDynamicQuery;
 import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.language.LanguageUtil;
 import com.liferay.portal.kernel.log.Log;
 import com.liferay.portal.kernel.log.LogFactoryUtil;
 import com.liferay.portal.kernel.model.Group;
@@ -60,6 +61,7 @@ import com.liferay.portal.kernel.util.ArrayUtil;
 import com.liferay.portal.kernel.util.GetterUtil;
 import com.liferay.portal.kernel.util.ListUtil;
 import com.liferay.portal.kernel.util.LocalizationUtil;
+import com.liferay.portal.kernel.util.StringBundler;
 import com.liferay.portal.kernel.util.StringUtil;
 import com.liferay.portal.kernel.util.Validator;
 import com.liferay.portal.kernel.workflow.WorkflowConstants;
@@ -71,6 +73,7 @@ import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.Set;
 
 import javax.portlet.PortletRequest;
 import javax.portlet.PortletResponse;
@@ -478,10 +481,33 @@ public class CPDefinitionIndexer extends BaseIndexer<CPDefinition> {
 			specificationOptionIds.add(
 				cpSpecificationOption.getCPSpecificationOptionId());
 
+			Set<Locale> availableLocales = LanguageUtil.getAvailableLocales(
+				cpDefinitionSpecificationOptionValue.getGroupId());
+
+			for (Locale locale : availableLocales) {
+				String languageId = LanguageUtil.getLanguageId(locale);
+
+				String localizedSpecificationOptionValue =
+					cpDefinitionSpecificationOptionValue.getValue(languageId);
+
+				if (Validator.isBlank(localizedSpecificationOptionValue)) {
+					localizedSpecificationOptionValue =
+						cpDefinitionSpecificationOptionValue.getValue(
+							cpDefinitionDefaultLanguageId);
+				}
+
+				document.addText(
+					StringBundler.concat(
+						languageId, "_SPECIFICATION_",
+						cpSpecificationOption.getKey(), "_VALUE_NAME"),
+					localizedSpecificationOptionValue);
+			}
+
 			document.addText(
 				"SPECIFICATION_" + cpSpecificationOption.getKey() +
 					"_VALUE_NAME",
-				cpDefinitionSpecificationOptionValue.getValue());
+				cpDefinitionSpecificationOptionValue.getValue(
+					cpDefinitionDefaultLanguageId));
 
 			document.addText(
 				"SPECIFICATION_" +

